src/components/screens/wallet/overview/accountInfo/identity.js

```diff
@@ -11,44 +25,74 @@ import styles from './accountInfo.css';
  *
  */
 const Identity = ({
-  address,
+  newAddress,
+  legacyAddress,
   bookmark,
   username = '',
+  t = str => str,
+  setShowLegacy,
 }) => {
-  if (!address) return null;
-  const truncatedAddress = truncateAddress(address);
-  const [isTruncated, setTruncationState] = useState(true);
-  const onClick = () => setTruncationState(!isTruncated);
+  if (!newAddress) {
+    return null;
```

Comment:
Its generally considered an anti pattern to have such a check inside the component. If we do this, the component will still be in the dom tree even if it returns `null`. I know we do this at other places as well but we have to slowly improve. Please let me know if you have a different POV. 👍

Reply:
I totally agree with you. In fact, we should hand this in the HOC level and never get here if the account doesn't exist. or fill it with a placeholder to gracefully handle loading state. but at this time, there's a chain of loading-not-handling components which cause this component to crash before the account is loaded. 
Given the due dates, I rather handle these issues with [this](https://github.com/LiskHQ/lisk-desktop/issues/3485), whenever we decide it's time.

Reply:
What I mean is, can't we do this in the render method of the parent? `newAddress && <Identity />`

Reply:
Sure. applying now